### DISCUSSION TITLE
Questions about the pipeline

### DISCUSSION THREAD

**Discussion by vwtlin** [compbiomed/singleCellTK#636] [Q&A]
Thanks very much for developing such a comprehensive tool for QC. I was using some of these tools individually before and really excited to see your paper. It is really handy to use! I was not clear about the best practices in using some of them. For example, should I remove the ambient RNA first before identifying doublets? In your pipeline, it looks like analysis using SoupX and DoubletFinder is done in parallel. 

I will be grateful if you could share your experience.

Cheers,
Weitao

**Comment by joshua-d-campbell** [comment]
Hi @vwtlin, thanks for trying out our tools! We generally recommend to run all cell QC tools on the original filtering matrix (i.e. the matrix with empty drops excluded) as this is what they were designed to do. That being said, sometimes QC'ing and filtering may need to be done in an iterative or step-wise fashion. For example, there have been a few single nucleus RNA-seq datasets where we first applied a filter for total counts/UMIs and features (e.g. total > 750 and detected > 500) before running the other ambient and doublet detection tools. We did this because we thought that CellRanger did not do a great job of distinguishing empty drops from true cells. I hope this helps but let me know if you have any specific questions on your data. I am going to move this to a Discussion since many people may have the same question.

**Reply by vwtlin** [reply]
Hi @joshua-d-campbell , thanks for sharing your experience. 

1. You mentioned that CellRanger didn't do great in some datasets. May I ask how you noticed that? Did you apply additional tools such as EmptyDrops to determine the optimal cut-off of UMI and gene counts?

2. Actually I don't have a very defined question on my own data. Briefly I observed cells spreading between clusters as outlined in the UMAP plot. The datasets came from 3 batches of single-cell RNA seq. They were processed by SoupX, followed by DoubletFinder. I am guessing the pipeline I used was not optimal therefore I am interested in asking for your experience using these QC tools. As you recommended in the other discussion, I'd better try DoubletFinder prior to DecontX.
(1) As suggested in [DoubletFinder tutorial](https://github.com/chris-mcginnis-ucsf/DoubletFinder), removing clusters with low RNA UMIs and high mito reads are recommended prior to the use of DoubletFinder. How are these filters are set in the singlecellTK pipeline? How to determine an optimal seuratRes (default as 1.5) in the `runDoubletFinder`?
(2) Following doublet removal, I would obtain the count matrix from the remaining cells and run DecontX. I am a little bit confused that this sounds contradictory to the standard workflow as you suggested in the other discussion (https://github.com/campbio/celda/discussions/375#discussioncomment-3307118). 

![UMAP](https://user-images.githubusercontent.com/44833405/196108258-f17ee295-571a-4a8a-b1e5-44f61feeca4b.png)

Cheers,
Weitao

**Comment by joshua-d-campbell** [comment]
Hi @vwtlin, 
1) We had very many nuclei (like 20K) when we were only expecting ~5K. With this many nuclei, it was a big "blob" on the UMAP so decontX and other QC tools didn't work well. So we applied a more stringent UMI filter to get better cells/nuclei. 

2) That is always a tricky question. What was the target number of cells loaded on the 10X? If it was high (like around 10K), then you may expect high numbers of doublets. If it was low, then it is not likely due to doublets (although you can see what different doublet tools say). This cluster could be a legitimate transitionary cell state, but you should be cautious if that is novel biology.

In terms of deciding on cutoffs, we generally look at the violin plot for each QC metric and decide a threshold that removes the lower or higher tails of the distribution. For deciding on clusters to use as input for DecontX, using a lower Seurat resolution is generally better to get fewer, broader clusters.

For a general procedure, I would suggest the following:
1) Run doublet detection and ambient RNA detection tools directly on the filtered matrix from CellRanger initially.
2) If there are too many cell/nuclei that look like a blob on the UMAP, then filter additional droplets out using a more stringent UMI cutoff and rerun the other QC tools. 
3) You can optionally remove high mito cells before running doublet detection tools, however we generally don't do this. 
4) We don't often filter out doublets unless they represent a substantial portion of the dataset and there are clear doublet-enriched clusters that can be removed. It is just good to flag potentially dubious clusters so you can keep track of that during your analysis.

**Reply by vwtlin** [reply]
@joshua-d-campbell Thanks so much for your comments. It took a while to get back to you as I was occupied by some wet lab recently. 

After trying different QC filters, SoupX and DecontX with different orders, the results did not get improved. Eventually considering the huge batch effects among three sets of data, I used Harmony to correct it and it worked quite well. All the clusters were separated quite well and now are moving on to more downstream analysis.

Hopefully once I finish all the downstream analysis, I could know if the Harmony workflow makes sense or not.

Cheers,
Weitao

**Comment by joshua-d-campbell** [comment]
Hi @vwtlin, thanks for following up. That makes sense. The QC tools can help improved the quality of the data and ensure that some clustering results aren't driven by certain types of technical artifacts. However, these tools do not handle batch effects between datasets which needs to be performed after filtering. Good luck!
